content/en/blog/2023/creating-k8s-cluster-with-runtime-observability.md

```diff
@@ -0,0 +1,472 @@
+Collector, like this:
+
+```bash
+Span #90
+    Trace ID       : 03a7bf9008d54f02bcd4f14aa5438202
+    Parent ID      :
+    ID             : d7a10873192f7066
+    Name           : KubernetesAPI
+    Kind           : SPAN_KIND_SERVER
+    Start time     : 2023-05-18 01:51:44.954563708 +0000 UTC
+    End time       : 2023-05-18 01:51:44.957555323 +0000 UTC
+    Status code    : STATUS_CODE_UNSET
+    Status message :
+Attributes:
+     -> net.transport: STRING(ip_tcp)
+     -> net.peer.ip: STRING(127.0.0.1)
+     -> net.peer.port: INT(54678)
+     -> net.host.ip: STRING(127.0.0.1)
+     -> net.host.port: INT(6443)
+     -> http.target: STRING(/api/v1/namespaces/kube-system/pods/helm-install-traefik-crd-8w4wd)
+     -> http.server_name: STRING(KubernetesAPI)
+     -> http.user_agent: STRING(k3s/v1.27.1+k3s1 (linux/amd64) kubernetes/bc5b42c)
+     -> http.scheme: STRING(https)
+     -> http.host: STRING(127.0.0.1:6443)
+     -> http.flavor: STRING(2)
+     -> http.method: STRING(GET)
+     -> http.wrote_bytes: INT(4724)
+     -> http.status_code: INT(200)
+```
+
+### Testing the Cluster Runtime
+
+With the Observability environment and the Kubernetes cluster set up, you can
+now trigger commands against Kubernetes and see traces of these actions in
+Jaeger.
+
+Open the browser, and navigate to the Jaeger UI located at
+[http://localhost:16686/search](http://localhost:16686/search). You’ll see that
+the `apiserver`, `containerd`, and `kubelet` services are publishing traces:
+
+![Jaeger screen with services dropdown open showing apiserver, containerd and kubelet services as options](https://res.cloudinary.com/djwdcmwdz/image/upload/v1684855720/Blogposts/creating-kubernetes-cluster-with-observability/screely-1684855324762_kk25pv.png)
+
+Choose `apiserver` and click on **"Find Traces”.** Here you see traces from the
+Kubernetes control plane:
+
+![Jaeger screen showing a list of spans found for apiserver](https://res.cloudinary.com/djwdcmwdz/image/upload/v1684855720/Blogposts/creating-kubernetes-cluster-with-observability/screely-1684855412765_xe7rnr.png)
+
+Let’s run a sample command against Kubernetes with `kubectl`, like running an
+echo:
+
+```bash
+kubectl run -it --rm --image=alpine echo-command -- echo Hi
+
+# Output
+# If you don't see a command prompt, try pressing enter.
+# warning: couldn't attach to pod/echo-command, falling back to streaming logs: unable to upgrade connection: container echo-command not found in pod echo-command_default
+# Hi
+# pod "echo-command" deleted
+```
+
+And now, open Jaeger again, choose the `kubelet` service, operation `syncPod`,
+and add the tag `k8s.pod=default/echo-command`, you should be able to see spans
+related to this pod:
+
+![Jaeger screen showing a list of spans found for the syncPod operation on kubelet service](https://res.cloudinary.com/djwdcmwdz/image/upload/v1684855720/Blogposts/creating-kubernetes-cluster-with-observability/screely-1684855456007_flekwn.png)
```

Comment:
Images should be local. For help on how to set that up see: https://github.com/open-telemetry/opentelemetry.io/blob/main/README.md#fork--write-locally